### ISSUE TITLE
C++17-friendly ?

### ISSUE THREAD

**Issue by Alexandre-Dhalenne** [danielaparker/jsoncons#335]
Would it be possible to make this library compliant with C++17 ? I'm getting a lot of 

```
[...]/jsoncons-0.167.1/include/jsoncons_ext/cbor/cbor_detail.hpp:29:42: error: binary constants are a C++14 feature or GCC extension [-Werror]
 const uint8_t cbor_array_tags_010_mask = 0b11100000;
```

And as in my current project I'm treating warning as errors I can not use the latest version of jsoncons. 

Thanks,

**Comment by danielaparker** [comment]
Thanks for pointing that out, the 0b... notation used is only valid with C++14 and later. I would expect that to be valid if compiling with C++17, though. In practice many C++11 compilers also support it, which is why we missed it in unit tests, even with warnings treated as errors.

We'll change it to hexadecimal notation in the next release, which should be soon.

**Comment by Alexandre-Dhalenne** [comment]
Great ! 
Again, thanks a lot for your library and fast support. 
Have a great day,

**Comment by danielaparker** [comment]
The candidate release is currently on master, which should fix the issue you reported.

**Comment by danielaparker** [comment]
Fixed in 0.168.0
